docs/deployment_userinterface.md

```diff
@@ -0,0 +1,302 @@
+# Deployment of User Interface
+This section provides information regarding the deployment of the BrainKB UI, both in the development and the production mode.
+
+## System Requirements
+- The BrainKB UI is based on NextJS; we need to install the Node.js. When writing this document, the minimum version is **Node.js 18.17 or later**. For more (or latest) information, please check NextJS website.
+
+	- NextJS: [https://nextjs.org/docs/getting-started/installation](https://nextjs.org/docs/getting-started/installation)
+	- **<span style="color: red;">Note:</span>** Manual installation of latest version of NextJS.
+	
+		```
+		npm install next@latest react@latest react-dom@latest
+		```
+		If you get issue with depenency, add `--legacy-peer-deps` options.
+	- Installation of the dependencies.
+		```
+		npm install 
+		```
+
+- Once all the NextJS requirements has been met or covered, next step is to configure the OAuth. In our case we use Google and GitHub. Therefore, you would need to obtain the OAuth client ID and secret and update the ```.env.local``` file (shown below). Additionally, you also need to provide the secret for [NextAuth](https://next-auth.js.org/), a library that we use authentication. The ```NEXTAUTH_URL``` for development is ```http://localhost:3000```. Similarly, the GitHub secret and client ID can be obtained from GitHub at [https://github.com/settings/tokens](https://github.com/settings/tokens) and Google at [https://console.cloud.google.com/](https://console.cloud.google.com/).
```

Comment:
```suggestion
- Once all the NextJS requirements has been met or covered, next step is to configure the OAuth, and create `.env.local` file (shown below) in the current directory.

 In our case we use Google and GitHub. Therefore, you would need to obtain the OAuth client ID and secret and update the file. Additionally, you also need to provide the secret for [NextAuth](https://next-auth.js.org/), a library that we use authentication. The ```NEXTAUTH_URL``` for development is ```http://localhost:3000```. Similarly, the GitHub secret and client ID can be obtained from GitHub at [https://github.com/settings/tokens](https://github.com/settings/tokens) and Google at [https://console.cloud.google.com/](https://console.cloud.google.com/).
```

Reply:
it's not clear to me what i should do to obtain all the information, we can talk later today, and perhaps we can clarify together